src/__test__/connection/dns.test.ts

```diff
@@ -0,0 +1,49 @@
+import { optionalDescribe } from "../utils";
+
+import { EventStoreDBClient, jsonEvent } from "../..";
+
+optionalDescribe(process.env.EVENTSTORE_CLOUD_ID != null)(
+  "dns discover",
+  () => {
+    const STREAM_NAME = "test_stream_name";
+    const { EVENTSTORE_CLOUD_ID } = process.env;
+    const event = jsonEvent({
+      eventType: "test",
+      payload: { message: "test" },
+    });
+
+    describe("should successfully connect", () => {
+      test("connectionString", async () => {
+        const client = EventStoreDBClient.connectionString`esdb+discover://admin:changeit@${EVENTSTORE_CLOUD_ID!}.mesdb.eventstore.cloud?nodePreference=leader`;
```

Comment:
**SUGGESTION:** It might be worth checking both node preferences (leader and follower) for two types of client setups (currently leader is only tested with connection string and options for follower). This could be setup by wrapping describe with foreach on ["leader", "follower"] and doing simple tests matrix.

Reply:
Yeah, these tests could probably be expanded on. We're not actually testing if it connects to a leader or follower, just that it connects. These are tested elsewhere (connection string parsing, connecting to specifed nodePreference), but we should probably test that its all hooked up correctly on the DNS code path.